pico-statsd/src/main/scala/org/pico/statsd/PicoStatsDClient.java

```diff
@@ -289,9 +283,9 @@ public PicoStatsDClient(final String prefix, final String hostname, final int po
      *     if the client could not be started
      */
     public PicoStatsDClient(final String prefix,  final int queueSize, String[] constantTags, final StatsDClientErrorHandler errorHandler,
-                                   final Callable<InetSocketAddress> addressLookup) throws StatsDClientException {
+                            final Callable<InetSocketAddress> addressLookup) throws StatsDClientException {
         if((prefix != null) && (!prefix.isEmpty())) {
-            this.prefix = String.format("%s.", prefix);
+            this.prefix = "%s." + prefix;
```

Comment:
All the performance changes fit this pattern.  Use of `String.format` is replaced with string concatenation because the latter is faster.  It would be good to get all such changed eye-balled to ensure there are no typos.